Tests/IGListAdapterTests.m

```diff
@@ -1067,6 +1067,24 @@ - (void)test_whenSupplementarySourceReturnsNegativeSize_thatAdapterReturnsZero {
     XCTAssertEqual(size.height, 0.0);
 }
 
+- (void)test_whenSectionIndexGreaterThanZero_thatAdapterReturnsSizeForSectionIndexesGreaterThanZero {
+  self.dataSource.objects = @[@1];
+  [self.adapter reloadDataWithCompletion:nil];
+  
+  IGTestSupplementarySource *supplementarySource = [IGTestSupplementarySource new];
+  supplementarySource.collectionContext = self.adapter;
+  supplementarySource.supportedElementKinds = @[UICollectionElementKindSectionFooter];
+  
+  IGListSectionController *controller = [self.adapter sectionControllerForObject:@1];
+  controller.supplementaryViewSource = supplementarySource;
+  supplementarySource.sectionController = controller;
+  
+  const CGSize size = [self.adapter sizeForSupplementaryViewOfKind:UICollectionElementKindSectionHeader
+                                                       atIndexPath:[NSIndexPath indexPathForItem:0 inSection:1]];
```

Comment:
@gobetti the way we use `IGListSupplementaryViewSource` in Instagram is to either:

- Create an object per `IGListSectionController` that conforms to the supplementary source protocol. Attach it as a delegate, inject as many dependencies as you want. @jessesquires might have more tips on this too.
- Just assign `self.supplementaryViewSource = self` inside the section controller and implement the required methods [like this](https://github.com/Instagram/IGListKit/blob/master/Examples/Examples-iOS/IGListKitExamples/SectionControllers/FeedItemSectionController.swift#L46-L65)

I'll admit you're definitely convincing me that this API could be updated though. I like the idea of creating a single object at the VC level that you can assign to individual section controllers, but there's no way to distinguish between section controllers.

Starting to think updating the API to include the object would be a nice update! Something like:

```
- (__kindof UICollectionReusableView *)viewForSupplementaryElementOfKind:(NSString *)elementKind
  withObject:(id)object
  atIndex:(NSInteger)index;

- (CGSize)sizeForSupplementaryViewOfKind:(NSString *)elementKind
  withObject:(id)object
  atIndex:(NSInteger)index;
```

What do you think?